### DISCUSSION TITLE
Failed to look up maven dependency androidx.***

### DISCUSSION THREAD

**Discussion by scottyab** [renovatebot/renovate#18456] [Request Help]
Hey, just starting my Renovate journey integrating it into an Android app project. I get Dependency Lookup Warnings for Google hosted packages: 

* Failed to look up maven dependency com.google.android.material:material
* Failed to look up maven dependency androidx.constraintlayout:constraintlayout
* Etc 

I believe these dependencies are from the maven repos `google()` Is there something additional I need to configure to pick up these `com.google.*` and `androidx.` dependencies?  Thanks! 

Also if there are any well know open-source projects using Renovate I'd love to have look at their `renovate.json` files.

**Comment by rarkins** [comment]
How are you configuring the `google()` repo? Do you have an example `pom.xml` you can share? We might need to add capabilities - I'm not sure.

Google the company happens to be a heavy user of Renovate so that might help you, e.g. https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fgithub.com%2Fgoogleapis+renovate and https://www.google.com/search?q=site%3Ahttps%3A%2F%2Fgithub.com%2Fgoogle+renovate

**Comment by scottyab** [comment]
> How are you configuring the google() repo?

Here's a [open source sample app](https://github.com/android/views-widgets-samples/blob/main/CardViewKotlin/build.gradle) which has the default Gradle repositories configured. 

I think it's the Android Gradle Plugin's DSL that resolves `google()` to `https://dl.google.com/dl/android/maven2/`  it works similar to `mavenCentral()` but those dependences are being picked up a-ok. 

I missed the keyword "Android" from my extra question... I meant...  Also if there are any well know open-source *android* projects using Renovate I'd love to have look at their renovate.json files.

P.s @rarkins sorry I'm new to github discussions too and failed to replied in a thread 🤦

**Reply by kyhule** [reply]
Did you ever resolve this issue? I am running into the same thing right now. The google maven repo is not getting discovered by renovate in the hosts list even though I am configuring the `google()` repo in my build.

**Comment by yschimke** [comment]
Similar issue as well

https://github.com/joreilly/Confetti/issues/315

> Renovate failed to look up the following dependencies: Failed to look up maven package androidx.activity:activity-compose,

App is configured with https://github.com/joreilly/Confetti/blob/main/settings.gradle.kts

```
pluginManagement {
    listOf(repositories, dependencyResolutionManagement.repositories).forEach {
        it.apply {
           google() {
                content {
                    includeGroupByRegex(".*google.*")
                    includeGroupByRegex(".*android.*")
                }
            }
```

**Reply by Churro** [reply]
renovate extracts dependencies without running the code. It can't handle scope functions, apply objects to receivers and doesn't instrument `listOf`, `forEach`, etc.

Better specify explicitly in which repositories you expect plugins and regular libs. It will also make dependency lookups faster.
